apps/dash-label-properties/app.py

```diff
@@ -0,0 +1,456 @@
+    value_array = np.array([0] + values)[masked_labels]
+    # Prefer a masked array so we can handle integer values
+    # Background cannot be zero when using datashader (because 'span' is not correctly implemented
+    # in current old version)
+    masked_values = np.ma.masked_where(~mask, value_array)
+
+    print("mode is", mode)
+    opacity = 0.4 if mode is None else 1
+
+    fig = px.imshow(img, binary_string=True, binary_backend="jpg",)
+    # Overlay the current labels
+    _, hex_list = zip(*colors.PLOTLY_SCALES["Viridis"])
+    label_data_array = xr.DataArray(np.flipud(masked_values))
+    label_multi_byte = ds.transfer_functions.shade(
+        label_data_array, cmap=list(hex_list)
+    ).to_bytesio()
+    label_multi_enc = base64.b64encode(label_multi_byte.getvalue()).decode()
+    label_multi_str = "data:image/png;base64,{}".format(label_multi_enc)
+
+    # Add the color value overlay
+    fig.add_image(colormodel="rgba", source=label_multi_str, opacity=opacity)
+    # Add a colorbar
+    fig.add_scatter(
+        x=[None],
+        y=[None],
+        mode="markers",
+        showlegend=False,
+        marker=dict(
+            colorscale=hex_list,
+            showscale=True,
+            cmin=-5,
+            cmax=5,
+            colorbar=dict(
+                thickness=0.05,
+                tickvals=[-5, 5],
+                ticktext=[f"{np.min(values[values!=0]):.2f}", f"{np.max(values):.2f}",],
+                len=0.6,
+                lenmode="fraction",
+                thicknessmode="fraction",
+                outlinewidth=1,
+                title=dict(text=f"<b>{color_column.capitalize()}</b>"),
+            ),
+        ),
+        hoverinfo="none",
+    )
+
+    # Disable the hover information for figure so far
+    fig.update_traces(hoverinfo="skip", hovertemplate=None)
+
+    # Overlay the colored label info with a hidden (opaque) scatter trace to display hoverinfo
+    # see also: https://scikit-image.org/docs/dev/auto_examples/segmentation/plot_regionprops.html#sphx-glr-auto-examples-segmentation-plot-regionprops-py
+    for rid, row in data_table.iterrows():
+        label = row.label
+        contour = measure.find_contours(label_array == label, 0.5)[0]
+        y, x = contour.T
+        hoverinfo = (
+            "<br>".join(
+                [
+                    # All numbers are passed as floats. If there are no decimals, cast to int for visibility
+                    f"{prop_name}: {f'{int(prop_val):d}' if prop_val.is_integer() else f'{prop_val:.3f}'}"
+                    if np.issubdtype(type(prop_val), "float")
+                    else f"{prop_name}: {prop_val}"
+                    for prop_name, prop_val in row[active_columns].iteritems()
+                ]
+            )
+            + " <extra></extra>"  # Add this to remove the trace name. See e.g. https://plotly.com/python/reference/#scatter-hovertemplate
+        )
+        fig.add_scatter(
+            x=x,
+            y=y,
```

Comment:
is the layout Image above really needed here, or would it be possible just to use the filled scatter traces with suitable colors computed from with datashader? Not very important since it works as it is but I'm just trying to see whether we can reduce the complexity of the function or not.

Reply:
Yes, I think that's a great idea. And I think we don't even need the datashader then anymore because we can simply use the matplotlib cmap object for "viridis" and call the rgb colors for each (normalized) value we want to draw individually. I have committed this and I think it's a lot cleaner and most of the complexity is gone.

Unfortunately I don't know how to integrate the colorbar-scatter into the "color overlay" scatter. So for the moment, we have these two, relatively lengthy scatter traces in there.